src/Columns/ColumnTuple.h

```diff
@@ -39,17 +44,16 @@ class ColumnTuple final : public COWHelper<IColumn, ColumnTuple>
     requires std::is_rvalue_reference_v<Arg &&>
     static MutablePtr create(Arg && arg) { return Base::create(std::forward<Arg>(arg)); }
 
+    static MutablePtr create(size_t len_) { return Base::create(len_); }
+
     std::string getName() const override;
     const char * getFamilyName() const override { return "Tuple"; }
     TypeIndex getDataType() const override { return TypeIndex::Tuple; }
 
     MutableColumnPtr cloneEmpty() const override;
     MutableColumnPtr cloneResized(size_t size) const override;
 
-    size_t size() const override
-    {
-        return columns.at(0)->size();
-    }
+    size_t size() const override { return len; }
```

Comment:
And consider adding more asserts to other places to ensure that `len` is consistent with columns' sizes.

```suggestion
    size_t size() const override
     {
         chassert(columns.empty() || columns.at(0)->size() == len);
         return len;
      }
```

Reply:
Done. This is indeed useful.